lisa/tools/git.py

```diff
@@ -70,19 +71,28 @@ def clone(
         full_path = cwd / code_dir
         self._log.debug(f"code path: {full_path}")
         if ref:
-            self.checkout(ref, cwd=full_path)
+            if branch:
+                self.checkout(ref, cwd=full_path, checkout_branch=branch)
+            else:
+                self.checkout(ref, cwd=full_path)
+        elif branch:
+            self.checkout("", cwd=full_path, checkout_branch=branch)
         return full_path
 
     def checkout(
         self, ref: str, cwd: pathlib.PurePath, checkout_branch: str = ""
     ) -> None:
-        if not checkout_branch:
-            # create a temp branch to checkout tag or commit.
-            checkout_branch = f"{constants.RUN_LOGIC_PATH}"
+        if ref:
+            if not checkout_branch:
+                # create a temp branch to checkout tag or commit.
+                checkout_branch = f"{constants.RUN_LOGIC_PATH}"
+            parameters = f"checkout {ref} -b {checkout_branch}"
+        else:
+            parameters = f"checkout {checkout_branch}"
```

Comment:
The `checkout_branch` is used to avoid modify local branches, it's the target name of branch. So it's a name the same as LISA run.

Reply:
This is just to use whenever `ref` is not provided. It will be useful as a simple checkout to a remote branch. For ex: `checkout master` without need of any ref.

Reply:
branch is a kind of ref.

Reply:
oh, but I'm running into errors related to git checkout while triggering legacy tests. I tried the same manually too, it didnt work. ` git checkout master -b random_name`
![image](https://user-images.githubusercontent.com/81740028/127045960-827bf579-7ab1-4745-9324-dc305074d6aa.png)

Reply:
I see, it needs the remote name too. It should be `origin/master`. Can you change the default name to include the origin?

Reply:
yes, it worked. Thanks @squirrelsc  I'll update `legacy_runner` to pass `origin/branch` as ref value.